### DISCUSSION TITLE
Troubleshooting step won't work for flatpak neovim

### DISCUSSION THREAD

**Discussion by cassepipe** [glacambre/firenvim#1645] [General]
I don't dare creating an issue so I am trying here first

> Make sure the firenvim script has been created
> 
> Running call firenvim#install(0) should have created a shell or batch script in $XDG_DATA_HOME/firenvim (on linux/osx, this usually is $HOME/.local/share/firenvim, on windows it's %LOCALAPPDATA%\firenvim). Make sure that the script exists and that it is executable. Try running it in a shell, like this:
> 
> echo 'abcde{}' | ${XDG_DATA_HOME:-${HOME}/.local/share}/firenvim/firenvim
> 
> This should print a json object the content of which is the current version of the firenvim neovim plugin. If it doesn't, please open a new github issue.

from https://github.com/glacambre/firenvim/blob/master/TROUBLESHOOTING.md#make-sure-the-neovim-plugin-is-installed

This step won't for troubleshooting because  XDG_DATA_HOME == ./var/app/io.neovim.nvim/data and is defined "inside" the flatpak so it's not available to the shell

But the following does print `true` :
```lua
flatpak run io.neovim.nvim --headless -c 'lua print(vim.fn.filereadable(vim.fn.expand("$XDG_DATA_HOME/firenvim/firenvim")) == 1)' 
```
So I suggest telling people to open neovim (so we don't have to care about what flaour they are using) and tell to paste the following *inside* their nvim instance instead :
```
lua print(vim.fn.filereadable(vim.fn.expand("$XDG_DATA_HOME/firenvim/firenvim")) == 1)
```
or something like that

What do you think ?

**Comment by glacambre** [comment]
Can Firenvim work with flatpak? The firenvim script that will be created in $XDG_DATA_HOME will directly reference the neovim binary which, if running flatpaks requires running `flatpak run io.neovim.nvim`, will not work.

**Comment by cassepipe** [comment]
Indeed I found the script and the reference to the nvim binary
The only solution I can think right now for flatpak, is to make a `nvim` wrapper script that just `exec flatpak run --env=NVIM_APPNAME=$NVIM_APPNAME $@` in $HOME/.local/bin but it feels like a hack

**Reply by cassepipe** [reply]
Or to modify the script directly of course but I guess this would overwritten by firenvim updates ?

**Comment by cassepipe** [comment]
@glacambre Would that be acceptable to modify the `.local/share/firenvim/firenvim` script to check for flatpak nvim too ?

```sh
if command -v nvim >/dev/null 2>/dev/null; then
  FIRENVIM_NVIM_BINARY=nvim
elif command -v flatpak >/dev/null 2>/dev/null; then
  FLATPAK_NVIM=$(flatpak list | grep -q 'neovim' | cut -f 2)
  if [ -n "$FLATPAK_NVIM" ]; then
    FIRENVIM_NVIM_BINARY="flatpak run --env=NVIM_APPNAME=$NVIM_APPNAME $FLATPAK_NVIM"
  else
    FIRENVIM_NVIM_BINARY=/usr/bin/nvim
  fi
fi
```

**Reply by glacambre** [reply]
I'm curious, what was the content of the `else` branch before you changed it?

**Reply by cassepipe** [reply]
I don't have it anymore but I am certain it was just
```sh
if command -v nvim >/dev/null 2>/dev/null; then
  FIRENVIM_NVIM_BINARY=nvim
else
  FIRENVIM_NVIM_BINARY=/usr/bin/nvim
fi
```

**Reply by cassepipe** [reply]
Found it ! : 
https://github.com/glacambre/firenvim/blob/64f9389b88c8b0c7667d45c171a5f25c42d852fb/autoload/firenvim.vim#L706C46-L707C38

**Reply by glacambre** [reply]
Yup, that's the piece of code that generates the script, I was just curious to see what it had generated for you. I don't understand why `/usr/bin/nvim` wouldn't work in your case. Did you run `firenvim#install()` from a neovim running in flatpack?

**Reply by cassepipe** [reply]
It's not that it wouldn't work it's just that I am about enabling the use of flatpak neovim

EDIT: Realize the snippet suggested above is needlessly complicated. Editing it right now.
